open-company/open-company-web
# Gif picker

#### Review after #858 

Card: https://trello.com/c/uUfVMXuv

Review with:
- [Infra](https://github.com/belucid/open-company-infrastructure/pull/106)
- [Lib](https://github.com/open-company/open-company-lib/pull/64)

NB(1): we need to create another API key on Giphy, or maybe another 2 for staging and production. For now we are using a test app i created but we need to request them a production key.


NB(2): Also this depends on the following PR and can not be merged:
https://github.com/cljsjs/packages/pull/1869
tests will be failing here until that gets merged in and release to cljsjs, but we can still deploy it to production if necessary installing the dev package there (like i did on staging, ping me for info).


To test:
- open cmail
- [x] can you add a gif
- publish
- [x] do you see the animated GIF in the expanded post?
- [x] do you NOT see the still version of the image in FoC?
- now open cmail again
- switch to fullscreen
- [x] add a GIF
- [x] add a photo
- [x] add a YouTube video
- publish
- [x] do you see the first image instead of the GIF on FoC?


### Before merging:
- lose the `-alpha` part of the version of open-company/open-company-lib in `project.clj`
- release that with `lein deploy release` (you need to be added to our organization in cljards first and then have your `CLOJARS_USER` and `CLOJARS_PASS` vars exported)
- change the lib version in all the services that use that version (only web in this case)
- commit lib and web changes
- merge both
- deploy

You should find more info on how to release a lib version in the lib README
